runtime/doc/api.txt

```diff
@@ -1969,6 +1969,10 @@ nvim_get_option_value({name}, {*opts})               *nvim_get_option_value()*
                 • win: |window-ID|. Used for getting window local options.
                 • buf: Buffer number. Used for getting buffer local options.
                   Implies {scope} is "local".
+                • filetype: |filetype|. Used to get the default option for a
+                  specific filetype. Cannot be used with any other option.
+                  Note: this is expensive, it is recommended to cache this
```

Comment:
I think we should be a bit more explicit here, e.g.,

"Note: This will trigger all filetype plugins and FileType autocommands for the corresponding filetype."

If more explanation is needed, it's probably better to add a short paragraph on the option to the main docstring body.